### PR TITLE
Fix break-separators=after-and-docked for lists and arrays

### LINKED ISSUE #835: Incomplete implementation of break-cases=after-and-docked
I believe currently the first field in the record behaves slightly differently from the rest:
```ocaml
let long_variable_name =
  { xxx = 1;
    yyy = 2;
    zzz = 3 }
in
```
Would it be possible to support a style where all fields get the same indentation? For example,
```ocaml
let long_variable_name =
  {
    xxx = 1;
    yyy = 2;
    zzz = 3
  }
in
```
Or a docked varaint (which I believe is already supported for record type declarations):
```ocaml
let long_variable_name = {
    xxx = 1;
    yyy = 2;
    zzz = 3
  }
in
```

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #835 and another step towards fixing #925.
Lists and arrays have to be tackled together because they both rely on the `fmt_expressions` function.
No regressions with test_branch, the diff below is actually a fix when `break-collection-expressions=wrap`:
```ocaml
diff --git a/sledge/src/import/import.ml b/sledge/src/import/import.ml
index 8380d79df..80613788c 100644
--- a/sledge/src/import/import.ml
+++ b/sledge/src/import/import.ml
@@ -118,8 +118,7 @@ module Invariant = struct
               (Error.create_s
                  (Base.Sexp.message "invariant failed"
                     [ ("", Source_code_position.sexp_of_t here)
-                    ; ("exn", sexp_of_exn exn)
-                    ; ("", sexp_of_t t) ])) in
+                    ; ("exn", sexp_of_exn exn); ("", sexp_of_t t) ])) in
           Caml.Printexc.raise_with_backtrace exn bt ) ;
       true )
 end
diff --git a/sledge/src/sledge_buck.ml b/sledge/src/sledge_buck.ml
index b7a05d911..306d6ba29 100644
--- a/sledge/src/sledge_buck.ml
+++ b/sledge/src/sledge_buck.ml
@@ -38,9 +38,8 @@ let buck_build ~context target =
   let open Process in
   eval ~context
     (run "buck"
-       [ "build"
-       ; "@mode/" ^ Lazy.force mode
-       ; "-c"; "sledge.build=True"; target ])
+       [ "build"; "@mode/" ^ Lazy.force mode; "-c"; "sledge.build=True"
+       ; target ])
 
 (* split a fully-qualified buck target into file and rule *)
 let parse_target target =
diff --git a/sledge/src/symbheap/exec.ml b/sledge/src/symbheap/exec.ml
index 8f9350764..68791079d 100644
--- a/sledge/src/symbheap/exec.ml
+++ b/sledge/src/symbheap/exec.ml
@@ -191,10 +191,8 @@ let memmov_up_spec us dst src len =
   {xs; foot; post}
 
 let memmov_specs us dst src len =
-  [ memmov_eq_spec us dst src len
-  ; memmov_dj_spec us dst src len
-  ; memmov_dn_spec us dst src len
-  ; memmov_up_spec us dst src len ]
+  [ memmov_eq_spec us dst src len; memmov_dj_spec us dst src len
+  ; memmov_dn_spec us dst src len; memmov_up_spec us dst src len ]
 
diff --git a/sledge/src/symbheap/solver.ml b/sledge/src/symbheap/solver.ml
index 7566ba70f..adfc38136 100644
--- a/sledge/src/symbheap/solver.ml
+++ b/sledge/src/symbheap/solver.ml
@@ -270,8 +270,7 @@ let excise_seg_sub_infix ({us; com; min; xs; sub; zs} as goal) msg ssg l_k
       (Exp.eq
          (Exp.memory ~siz:o ~arr:a)
          (Exp.concat
-            [| Exp.memory ~siz:l_k ~arr:a0
-             ; Exp.memory ~siz:n ~arr:a1
+            [| Exp.memory ~siz:l_k ~arr:a0; Exp.memory ~siz:n ~arr:a1
              ; Exp.memory ~siz:ko_ln ~arr:a2 |]))
       (Sh.star
          (Sh.seg {loc= k; bas= b; len= m; siz= l_k; arr= a0})
@@ -410,8 +409,7 @@ let excise_seg_min_infix ({us; com; min; xs; sub; zs} as goal) msg ssg k_l
          (Sh.and_
             (Exp.eq
                (Exp.concat
-                  [| Exp.memory ~siz:k_l ~arr:a0'
-                   ; Exp.memory ~siz:o ~arr:a
+                  [| Exp.memory ~siz:k_l ~arr:a0'; Exp.memory ~siz:o ~arr:a
                    ; Exp.memory ~siz:ln_ko ~arr:a2' |])
                (Exp.memory ~siz:n ~arr:a'))
```

### CODE REVIEW

File: src/Params.ml [hunk 1]
```diff
@@ -113,52 +113,112 @@ let get_record_type (c : Conf.t) =
...
-type record_expr =
+type elements_collection =
   { box: Fmt.t -> Fmt.t
-  ; break_after_with: Fmt.t
   ; sep_before: Fmt.t
   ; sep_after_non_final: Fmt.t
   ; sep_after_final: Fmt.t }
 
+type elements_collection_record_expr =
+  {common: elements_collection; break_after_with: Fmt.t}
```
Comment (Julow) [inline src/Params.ml]: I'm not fan of the `common` field. It is defined again for record pat. What do you think of removing it from `elements_collection_record_expr` and returning `elements_collection * elements_collection_record_expr` from `get_record_expr` ?

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -130,15 +127,29 @@ let function_indent c ~ctx ~default =
   | `Always, _ | _, (Top | Sig _ | Str _) -> c.conf.function_indent
   | _ -> default
 
-let fmt_expressions c width sub_exp exprs fmt fmt_expr =
+let fmt_expressions c width sub_exp exprs fmt_expr sep_before
+    sep_after_non_final sep_after_final =
```
Comment (Julow) [inline src/Fmt_ast.ml]: I think this function could take a `Params.elements_collection` argument instead.
It is always called like:
```ocaml
( p.box
    (fmt_expressions c ...
       p.sep_before p.sep_after_non_final p.sep_after_final)
```

Comment (gpetiot) [reply src/Fmt_ast.ml]: done

File: src/Params.ml (hunk 1 above)
Comment (gpetiot) [reply src/Params.ml]: done

**Pull request merged**